apps/mobile/src/libs/components/deadline-countdown/libs/hooks/use-countdown.hook.ts

```diff
@@ -0,0 +1,48 @@
+import { COUNTDOWN_EXPIRED } from "~/libs/constants/constants";
+import { useCallback, useEffect, useState } from "~/libs/hooks/hooks";
+import { type Countdown } from "~/libs/types/types";
+
+import { MillisecondsPerUnit } from "../../libs/enums/enums";
+import { getFormattedCountdownUnits } from "../../libs/helpers/helpers";
+
+const useCountdown = (deadline: string): Countdown => {
+	const [countdown, setCountdown] = useState<Countdown>(COUNTDOWN_EXPIRED);
+
+	const handleCalculateDaysUntilDeadline = useCallback((): boolean => {
```

Comment:
From the name of this helper, I expect that it should return a number, not a boolean
Either rename it to fit the return value, or split it into two separate helpers: one - to check if the deadline expires, second - to set the countdown values